src/FluentAssertions.AspNetCore.Mvc/PhysicalFileResultAssertions.cs

```diff
@@ -1,62 +1,185 @@
-﻿using FluentAssertions.Execution;
-using Microsoft.AspNetCore.Mvc;
-using System;
+﻿using System;
 using System.Diagnostics;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
 
 namespace FluentAssertions.AspNetCore.Mvc
 {
     /// <summary>
-    ///     Contains a number of methods to assert that a <see cref="PhysicalFileResult" /> is in the expected state.
+    /// Contains a number of methods to assert that a <see cref="PhysicalFileResult"/> is in the expected state.
     /// </summary>
     [DebuggerNonUserCode]
-    public class PhysicalFileResultAssertions : FileResultAssertions
+    public class PhysicalFileResultAssertions : ObjectAssertions
```

Comment:
I'm interested why did you separate the inheritance from the FileResultAssertions. (And also in the VirtualFileResultAssertion).
My reasoning was for inheritance is that PhysicalFileResult inherits from FileResult so the assertion also reusable.

Reply:
Ok. You are right. To be honest, I did not want to change your code.

Inheritance is a pretty cool thing. And the main purpose is to avoid "copy and paste".
And this is what I had to do a lot. (Pretty ugly).
But the idea of this framework is to use Method chaining.
And the Methods should be useable in any kind of Order.
This only works if the assertions have the exact return Value of the assertion.
Hence the SubClass and not the base class.
For example:
            controller.GetVirtualFileResult(TestFileName, testMediaTypeHeaderValue)
                .Should()
                .BeVirtualFileResult()
                .WithFileName(TestFileName)
                .WithContentType(testMediaTypeHeaderValue.ToString());

The WithFileName Method only exists in the subclass (VirtualFileResultAssertions), but not in the baseclass (FileResultAssertions).

Therefore this would not work, if I inherit from fileResultAssertions:
             controller.GetVirtualFileResult(TestFileName, testMediaTypeHeaderValue)
                .Should()
                .BeVirtualFileResult()
                .WithContentType(testMediaTypeHeaderValue.ToString());
                .WithFileName(TestFileName)

In your case, it seems to be a Bug, because you if you use at lease one Assertion Method of the FileResultAssertions class (baseclass) you cannot access any property or Method of the subclass. (e.g. FileName).

Therefore I suggest not to use inheritance here.

Reply:
You are right. I didn't realized this. In this case there is some classes that still isn't right: FileContentResultAssertions, FileStreamResultAssertions